File: internal/plugins/typescript/rules/ban_tslint_comment/ban_tslint_comment_test.go
package ban_tslint_comment

import (
	"testing"

	"github.com/web-infra-dev/rslint/internal/plugins/typescript/rules/fixtures"
	"github.com/web-infra-dev/rslint/internal/rule_tester"
)

func TestBanTslintCommentRule(t *testing.T) {
	rule_tester.RunRuleTester(fixtures.GetRootDir(), "tsconfig.json", t, &BanTslintCommentRule, []rule_tester.ValidTestCase{
		{Code: "let a: readonly any[] = [];"},
		{Code: "let a = new Array();"},
		{Code: "// some other comment"},
		{Code: "// TODO: this is a comment that mentions tslint"},
		{Code: "/* another comment that mentions tslint */"},
	}, []rule_tester.InvalidTestCase{
		{
			Code: "/* tslint:disable */",
			Errors: []rule_tester.InvalidTestCaseError{
				{MessageId: "commentDetected", Line: 1, Column: 1},
			},
			Output: []string{""},
		},
		{
			Code: "/* tslint:enable */",
			Errors: []rule_tester.InvalidTestCaseError{
				{MessageId: "commentDetected", Line: 1, Column: 1},
			},
			Output: []string{""},
		},
		{
			Code: "/* tslint:disable:rule1 rule2 rule3... */",
			Errors: []rule_tester.InvalidTestCaseError{
				{MessageId: "commentDetected", Line: 1, Column: 1},
			},
			Output: []string{""},
		},
		{
			Code: "/* tslint:enable:rule1 rule2 rule3... */",
			Errors: []rule_tester.InvalidTestCaseError{
				{MessageId: "commentDetected", Line: 1, Column: 1},
			},
			Output: []string{""},
		},
		{
			Code: "// tslint:disable-next-line",
			Errors: []rule_tester.InvalidTestCaseError{
				{MessageId: "commentDetected", Line: 1, Column: 1},
			},
			Output: []string{""},
		},
		{
			Code: "someCode(); // tslint:disable-line",
			Errors: []rule_tester.InvalidTestCaseError{
				{MessageId: "commentDetected", Line: 1, Column: 13},
			},
			Output: []string{"someCode();"},
		},
		{
			Code: "function f() {\n  return/* tslint:disable-line */foo;\n}",
			Errors: []rule_tester.InvalidTestCaseError{
				{MessageId: "commentDetected", Line: 2, Column: 9},
			},
			Output: []string{"function f() {\n  return foo;\n}"},
		},
		{
			Code: "// tslint:disable-next-line:rule1 rule2 rule3...",
			Errors: []rule_tester.InvalidTestCaseError{
				{MessageId: "commentDetected", Line: 1, Column: 1},
			},
			Output: []string{""},
		},
		{
			Code: "const whoa = doSomeStuff();\n// tslint:disable-line\nconsole.log(whoa);\n",
			Errors: []rule_tester.InvalidTestCaseError{
				{MessageId: "commentDetected", Line: 2, Column: 1},
			},
			Output: []string{"const whoa = doSomeStuff();\nconsole.log(whoa);\n"},
		},
		{
			Code: "/* tslint:disable */ const x = 1;",
			Errors: []rule_tester.InvalidTestCaseError{
				{MessageId: "commentDetected", Line: 1, Column: 1},
			},
			Output: []string{"const x = 1;"},
		},
	})
}

File: packages/rslint-test-tools/tests/typescript-eslint/rules/ban-tslint-comment.test.ts
import { RuleTester } from '@typescript-eslint/rule-tester';



const ruleTester = new RuleTester();

ruleTester.run('ban-tslint-comment', {
  valid: [
    {
      code: 'let a: readonly any[] = [];',
    },
    {
      code: 'let a = new Array();',
    },
    {
      code: '// some other comment',
    },
    {
      code: '// TODO: this is a comment that mentions tslint',
    },
    {
      code: '/* another comment that mentions tslint */',
    },
  ],
  invalid: [
    {
      code: '/* tslint:disable */',
      errors: [
        {
          column: 1,
          data: {
            text: '/* tslint:disable */',
          },
          line: 1,
          messageId: 'commentDetected',
        },
      ],
      output: '',
    },
    {
      code: '/* tslint:enable */',
      errors: [
        {
          column: 1,
          data: {
            text: '/* tslint:enable */',
          },
          line: 1,
          messageId: 'commentDetected',
        },
      ],
      output: '',
    },
    {
      code: '/* tslint:disable:rule1 rule2 rule3... */',
      errors: [
        {
          column: 1,
          data: {
            text: '/* tslint:disable:rule1 rule2 rule3... */',
          },
          line: 1,
          messageId: 'commentDetected',
        },
      ],
      output: '',
    },
    {
      code: '/* tslint:enable:rule1 rule2 rule3... */',
      errors: [
        {
          column: 1,
          data: {
            text: '/* tslint:enable:rule1 rule2 rule3... */',
          },
          line: 1,
          messageId: 'commentDetected',
        },
      ],
      output: '',
    },
    {
      code: '// tslint:disable-next-line',
      errors: [
        {
          column: 1,
          data: {
            text: '// tslint:disable-next-line',
          },
          line: 1,
          messageId: 'commentDetected',
        },
      ],
      output: '',
    },
    {
      code: 'someCode(); // tslint:disable-line',
      errors: [
        {
          column: 13,
          data: {
            text: '// tslint:disable-line',
          },
          line: 1,
          messageId: 'commentDetected',
        },
      ],
      output: 'someCode();',
    },
    {
      code: '// tslint:disable-next-line:rule1 rule2 rule3...',
      errors: [
        {
          column: 1,
          data: { text: '// tslint:disable-next-line:rule1 rule2 rule3...' },
          line: 1,
          messageId: 'commentDetected',
        },
      ],
      output: '',
    },
    {
      code: `
const whoa = doSomeStuff();
// tslint:disable-line
console.log(whoa);
      `,
      errors: [
        {
          column: 1,
          data: {
            text: '// tslint:disable-line',
          },
          line: 3,
          messageId: 'commentDetected',
        },
      ],
      output: `
const whoa = doSomeStuff();
console.log(whoa);
      `,
    },
  ],
});
